tests/Sharding/ShardDistributionReporterTest.cpp

```diff
@@ -1570,108 +1245,72 @@ TEST_F(ShardDistributionReporterTest,
-                    std::shared_ptr<ClusterCommCallback> callback, ClusterCommTimeout timeout,
-                    bool singleRequest, ClusterCommTimeout initTimeout) -> OperationID {
-        EXPECT_TRUE(path == "/_db/UnitTestDB/_api/collection/" + s1 + "/count");
-
-        OperationID opId = TRI_NewTickServer();
-        coordId = coordTransactionID;
-
-        ClusterCommResult response;
-        response.coordTransactionID = coordTransactionID;
-        response.operationID = opId;
-        response.answer_code = rest::ResponseCode::OK;
-        response.status = CL_COMM_RECEIVED;
-
-        if (destination == "server:" + dbserver1) {
-          response.result =
-              std::shared_ptr<SimpleHttpResult>(&lCount, [](SimpleHttpResult*) {});
-          leaderRes = response;
-        } else if (destination == "server:" + dbserver2) {
-          response.result =
-              std::shared_ptr<SimpleHttpResult>(&f1Count, [](SimpleHttpResult*) {});
-          follower1Res = response;
-        } else if (destination == "server:" + dbserver3) {
-          response.status = CL_COMM_TIMEOUT;
-          response.result =
-              std::shared_ptr<SimpleHttpResult>(&f2Count, [](SimpleHttpResult*) {});
-          follower2Res = response;
-        } else {
-          EXPECT_TRUE(false);
-        }
-
-        return opId;
-      });
-  fakeit::When(Method(commMock, wait))
-      .AlwaysDo([&](CoordTransactionID const coordTransactionID, OperationID const operationID,
-                    ShardID const& shardID, ClusterCommTimeout timeout) {
-        if (operationID != 0) {
-          return leaderRes;
-        }
-        if (returnedFirstFollower) {
-          return follower2Res;
-        } else {
-          returnedFirstFollower = true;
-          return follower1Res;
-        }
-      });
+  // Mocking sendRequest for count calls
+  auto sender = [&, this](network::DestinationId const& destination,
+                          arangodb::fuerte::RestVerb reqType, std::string const& path,
+                          velocypack::Buffer<uint8_t> body, network::Timeout timeout,
+                          network::Headers headerFields) -> network::FutureRes {
+    EXPECT_TRUE(path == "/_db/UnitTestDB/_api/collection/" + s1 + "/count");
+
+    network::Response response;
+    response.destination = destination;
+    response.error = fuerte::Error::NoError;
+
+    if (destination == "server:" + dbserver1) {
+      response.response =
+          std::shared_ptr<fuerte::Response>(&lCount, [](fuerte::Response*) {});
+    } else if (destination == "server:" + dbserver2) {
+      response.response =
+          std::shared_ptr<fuerte::Response>(&f1Count, [](fuerte::Response*) {});
+    } else if (destination == "server:" + dbserver3) {
+      response.error = fuerte::Error::Timeout;
+      response.response =
+          std::shared_ptr<fuerte::Response>(&f2Count, [](fuerte::Response*) {});
```

Comment:
fuerte won't create a response object if there was an error like Timeout
```suggestion
          nullptr;
```